syncopy/tests/backend/test_conn.py

```diff
@@ -236,15 +234,13 @@ def test_regularization():
         A = np.random.randn(50)
         CSD += np.outer(A, A)
 
-    # get CSD condition number, which is way too large!
-    CN = np.linalg.cond(CSD).max()
-    print(CN)
-    cmax = 1e6
-    assert CN > cmax
-
     # --- regularize CSD ---
 
-    CSDreg, fac = regularize_csd(CSD, cond_max=cmax, nSteps=25)
+    cmax = 1e4
+    CSDreg, fac, iniCN = regularize_csd(CSD, cond_max=cmax, nSteps=25)
+    # get initial CSD condition number, which is way too large!
+    print(iniCN)
+    assert iniCN > cmax
```

Comment:
would be nice to use 2nd arg of assert statement to print better error here, like

   assert iniCN > cmax, f"some message about var {var} here"

Reply:
yes, I've seen this only in your code now.. did not know this is possible, but good idea :+1:

Reply:
puh, there was definitely something off in any case.. I corrected it, still not sure how often ill conditioned CSDs appear in the wild :sweat_smile: 
But at least now we know at least _aposteriori_ what the factorization/regularization did thanks to the additional cF returns :)